### ISSUE TITLE
Incrementally defining structures with backtick-escaped names does not work well

### ISSUE THREAD

**Issue by alexrp** [Vector35/binaryninja-api#3192]
**Version and Platform (required):**
- Binary Ninja Version: 3.1.3504-dev
- OS: Windows
- OS Version: 10.0.25136.1000
- CPU Architecture: x64

**Bug Description:**
BN seems to have trouble dealing with custom structure types that use backtick-escaped names.

**Steps To Reproduce:**
Try to define the following types:

```c
struct `PbPuppet::Packet`;

struct `PbPuppet::Packet__fields` __packed
{
    int64_t field_1;
    int32_t field_2;
    uint8_t* field_3;
    uint8_t* field_4;
    int32_t field_5;
};

struct `PbPuppet::Packet__vtable` __packed
{
    void (* __slot1)(struct `PbPuppet::Packet`* `this`);
    void (* __slot2)(struct `PbPuppet::Packet`* `this`);
    void (* __slot3)(struct `PbPuppet::Packet`* `this`);
    void (* __slot4)(struct `PbPuppet::Packet`* `this`);
    void (* __slot5)(struct `PbPuppet::Packet`* `this`);
    void (* __slot6)(struct `PbPuppet::Packet`* `this`);
    void (* __slot7)(struct `PbPuppet::Packet`* `this`);
    void (* __slot8)(struct `PbPuppet::Packet`* `this`);
    void (* __slot9)(struct `PbPuppet::Packet`* `this`);
    void (* __slot10)(struct `PbPuppet::Packet`* `this`);
    void (* __slot11)(struct `PbPuppet::Packet`* `this`);
    void (* __slot12)(struct `PbPuppet::Packet`* `this`);
};

struct `PbPuppet::Packet` __packed
{
    struct `PbPuppet::Packet__vtable`* __vtable;
    struct `PbPuppet::Packet__fields` __fields;
};
```

**But**, do it incrementally, rather than pasting the full definitions into the Create New Types dialog (the latter approach works fine).

* Try defining the vtable with `this` pointers typed as `void*` first, then try to change its type to `PbPuppet::Packet*`. You'll find you can't refer to it without the `struct´ keyword which you normally can.
* Try defining `PbPuppet::Packet` as a structure with size `0x28`, then fill in `__vtable` and `__fields` by adding fields manually. You should see the fields being added as empty anonymous structures.
* Possibly other issues I forgot? This area seems to be a bit rough in general.

**Comment by alexrp** [comment]
Also, even if you do paste the above types into Create New Types in one go, trying to change a variable's type to `PbPuppet::Packet` does not work out well:

![image](https://user-images.githubusercontent.com/44076/173239909-607cea12-ddb6-4d9f-b998-eeb08dcbaf55.png)

**Comment by CouleeApps** [comment]
Can confirm, this looks like a bug in the new clang type resolver with escaped names (it tries to find the type \`PbPuppet::Packet\` instead of PbPuppet::Packet)

**Comment by CouleeApps** [comment]
This will be fixed in the latest build, 3.1.3522-dev
